### ISSUE TITLE
There are incompatible versions in the resolved dependencies yet pip install works

### ISSUE THREAD

**Issue by mambon2** [jazzband/pip-tools#1604]
Here is the constraint you can pass to pip-compile
apache-airflow[crypto,celery,mysql,redis]

pip install will gladly install without conflict. Yet pip-compile throws this error:

`Could not find a version that matches click<8.0,<9.0,>=4.0,>=5.1,>=6.7,>=8.0.3 (from flask==1.1.4->apache-airflow[celery,crypto,mysql,redis]==2.2.4->-r requirements.in (line 1))
Tried: 0.1, 0.1, 0.2, 0.2, 0.3, 0.3, 0.4, 0.4, 0.5, 0.5, 0.5.1, 0.5.1, 0.6, 0.6, 0.7, 0.7, 1.0, 1.0, 1.1, 1.1, 2.0, 2.0, 2.1, 2.1, 2.2, 2.2, 2.3, 2.3, 2.4, 2.4, 2.5, 2.5, 2.6, 2.6, 3.0, 3.0, 3.1, 3.1, 3.2, 3.2, 3.3, 3.3, 4.0, 4.0, 4.1, 4.1, 5.0, 5.0, 5.1, 5.1, 6.0, 6.0, 6.1, 6.1, 6.2, 6.2, 6.3, 6.3, 6.4, 6.4, 6.5, 6.6, 6.6, 6.7, 6.7, 7.0, 7.0, 7.1, 7.1, 7.1.1, 7.1.1, 7.1.2, 7.1.2, 8.0.0, 8.0.0, 8.0.1, 8.0.1, 8.0.2, 8.0.2, 8.0.3, 8.0.3, 8.0.4, 8.0.4, 8.1.0, 8.1.0
Skipped pre-versions: 6.7.dev0, 8.0.0a1, 8.0.0a1, 8.0.0rc1, 8.0.0rc1
There are incompatible versions in the resolved dependencies:
  click>=4.0 (from clickclick==20.10.2->apache-airflow[celery,crypto,mysql,redis]==2.2.4->-r requirements.in (line 1))
  click<9.0,>=8.0.3 (from celery==5.2.3->apache-airflow-providers-celery==2.1.3->apache-airflow[celery,crypto,mysql,redis]==2.2.4->-r requirements.in (line 1))
  click<9,>=6.7 (from flask-appbuilder==3.4.4->apache-airflow[celery,crypto,mysql,redis]==2.2.4->-r requirements.in (line 1))
  click<8.0,>=5.1 (from flask==1.1.4->apache-airflow[celery,crypto,mysql,redis]==2.2.4->-r requirements.in (line 1))
`

#### Environment Versions

1. Ubuntu 18.04
1. Python 3.10.3
1. pip 22.0.4
1. pip-compile, version 6.5.1

#### Steps to replicate

See description above

#### Expected result

pip-compile should resolve dependencies the same way pip install does for this constraint.

#### Actual result

See description above

**Comment by AndydeCleyre** [comment]
Yes, I can reproduce, thanks!

Good news: this succeeds when using #1539 with the new resolver option.

e.g. with [pipx](https://github.com/pypa/pipx):

```console
$ pipx run --spec 'pip-tools @ git+https://github.com/atugushev/pip-tools@new-resolver' pip-compile --resolver backtracking 
```

or with [zpy](https://github.com/AndydeCleyre/zpy):

```console
% pipz runpkg 'pip-tools @ git+https://github.com/atugushev/pip-tools@new-resolver' pip-compile --resolver backtracking 
```

**Comment by mambon2** [comment]
Thanks @AndydeCleyre when can we expect this fix to be released?

**Comment by mambon2** [comment]
This seems to take a lot longer to execute.  Is this normal?

**Comment by mambon2** [comment]
Thanks Andy, can you tell me which release number this is slated to be in?

**Comment by AndydeCleyre** [comment]
@mambon2 

I don't know, maybe @atugushev has an idea.

There are just a couple things remaining to be ironed out.

**Comment by atugushev** [comment]
This has been fixed in #1539 with the backtracking resolver, try `pip-compile --resolver backtracking`. The resolver is released as part of pip-tools `v6.8.0`. Please let us know if it doesn't resolve your issue. Thanks!

**Comment by adminy** [comment]
In my case, I removed `requirements.txt` and just left `requirements.in` then I ran `pip-compile` again and that fixed the issue. Setting the resolver was not enough ...
